crates/nu-path/src/helpers.rs

```diff
@@ -2,12 +2,22 @@
 use omnipath::WinPathExt;
 use std::path::PathBuf;
 
-pub fn home_dir() -> Option<PathBuf> {
-    dirs_next::home_dir()
-}
+pub use home::home_dir;
 
 pub fn config_dir() -> Option<PathBuf> {
-    dirs_next::config_dir()
+    std::env::var("XDG_CONFIG_HOME")
+        .ok()
+        .and_then(|path| {
+            let path = PathBuf::from(path);
+
+            // Return None if the path obtained from the environment variable isn’t absolute.
```

Comment:
I don't _think_ we should be returning None if the environment variable isn't set right. I suspect that will make things fail with rather unhelpful error messages (but I haven't tested that).

Maybe it would make more sense for `config_dir()` to return a Result instead of an Option?

Reply:
ping @utkarshgupta137

Before we land this I think we need to ensure that Nushell generates useful errors if `XDG_CONFIG_HOME` is set to something invalid

Reply:
Maybe?

My primary concern is that Nushell should behave well (probably: exit with a helpful error message) when given an `XDG_CONFIG_HOME` value that it can't handle. I think that will most likely involve returning a `Result` but I'm open to any approach that works.

Reply:
That might work, as long as we 1) provide a useful error message 2) don't end up in a bad state. I'm currently not sure what happens when this function returns `None`.

Reply:
I've gone through the places where this function is called & 3 outcomes:
- User is prompted to create the folder
- Error handling already exists
- Error handling is not required (for eg. reading history or loading plugins)

So, I've added an eprintln statement if the `XDG_CONFIG_HOME` is not absolute.

Reply:
Looks good to me, thanks. I think I was misunderstanding the code before.